extend/generic-writer/configuration/configuration.md

```diff
@@ -0,0 +1,718 @@
+
+```json
+{
+  "data": [
+  {"user": 
+     {"id": 1}
+   },
+  {"user": 
+     {"id": 2}
+   }
+  ]
+}
+```
+
+See examples: [012](https://bitbucket.org/kds_consulting_team/kds-team.wr-generic/src/master/docs/examples/012-simple-json-request-data-wrapper/)
+
+##### Column names override
+
+You may override specific column names using the `column_names_override` parameter 
+to be able to generate fields with characters not supported in Storage column names.
+
+**NOTE** that this is applied **after** the column type definition, so refer to original name in the `column_types` config. 
+
+**NOTE2** It is possible to rename nested objects as well. The rename is applied to the leaf node. E.g. `"address___city":"city.address"` 
+with delimiter set to `___` will result in `{"address":{"city.address":"SOME_VALUE"}}`. See [example 23](https://bitbucket.org/kds_consulting_team/kds-team.wr-generic/src/master/docs/examples/023-simple-json-nested-object-rename-column/)
+
+**Example:**
+
+```json
+"request_options": {
+            ...,
+            "content": {
+                            "content_type": "JSON",
+                            "json_mapping": {
+                                "nesting_delimiter": "_",
+                                "chunk_size": 1,
+                                "column_names_override": {
+                                          "field_id": "field-id",
+                                           "full_name": "FULL.NAME"     
+                                      }
+```
+
+For more details refer to examples: [20](https://bitbucket.org/kds_consulting_team/kds-team.wr-generic/src/master/docs/examples/020-simple-json-column-name-override/)
+and [23](https://bitbucket.org/kds_consulting_team/kds-team.wr-generic/src/master/docs/examples/023-simple-json-nested-object-rename-column/)
+
+
+#### Iterate By Columns
+
+This parameter allows performing the requests in iterations based on provided parameters within data. The user specifies
+columns in the source table that will be used as parameters for each request. The column values may be then used instead of placeholders 
+within the `request_options`. The input table is iterated row by row, e.g. 1 row = 1 request.
+
+
+```json
+"request_options": {
+    ...,
+    "content": {
+        "content_type": "JSON",
+        "iterate_by_columns": [
+        "id", "date"
+        ]
+    }
+}
+```
+
+These will be injected in:
+
+- `api_request.endpoint_path` if placeholder is specified, e.g.  `/user/[[id]]`
+- `user_parameters` section, any existing parameters with a same name will be replaced by the value from the data. This
+  allows for example for changing request parameters dynamically `www.example.com/api/user?date=xx` where the `date` value is specified
+  like:
+
+```json
+"request_options": {
+    "api_request": {
+        "method": "POST",
+        "endpoint_path": "/customer/[[id]]",
+        "query_parameters": {
+              "date": {
+                      "attr": "date"
```

Comment:
so this needs a user_parameters section to be complete, does it? I'm sure it's clever, but I don't quite get it here :)